client/src/pages/ChatsListPage.tsx

```diff
@@ -1,80 +1,87 @@
-import React, { useEffect } from 'react';
+import { useState, useEffect } from 'react';
+import { Chat } from '../firebase/chat';
 import { ChatItem } from '../components/ChatItem';
-import { useCollectionData } from 'react-firebase-hooks/firestore';
-import { auth, collections } from '../firebase/connection';
-import { orderBy, query, where } from 'firebase/firestore';
+import { auth } from '../firebase/connection';
 import { useAuthState } from 'react-firebase-hooks/auth';
 import { NavLink, useNavigate } from 'react-router-dom';
+import { Role, getRealtimeUserChats, assignSupporter } from '../helpers/chatFunctions';
 import Button from '../components/Button';
-import { assignSupporter } from '../helpers/chatFunctions';
-import SupporterWaiting from '../components/SupporterWaiting';
 
 export const ChatsListPage = () => {
+  const navigate = useNavigate();
   const [user, userLoading] = useAuthState(auth);
-  const [chats, chatsLoading] = useCollectionData(
-    query(
-      collections.chats,
-      where('supporterId', '==', userLoading || !user ? 'empty' : user?.uid),
-      orderBy('createdAt', 'desc')
-    )
-  );
+  const [chats, setChats] = useState<Chat[]>([]);
+
+  const activeChats = chats?.filter((chat) => chat.status != "ended");
+  const endedChats = chats?.filter((chat) => chat.status == "ended");
 
-  const navigate = useNavigate();
 
   useEffect(() => {
+    // redirect if user not logged in
     if (!userLoading && !user) navigate('/');
   }, [user, userLoading, navigate]);
 
-  const activeChats = chats?.filter((chat) => chat.status === 'active');
-  const endedChats = chats?.filter((chat) => chat.status === 'ended');
-
-  let mainContent = (
-    <>
-      <h2 className="header">רשימת נתמכים</h2>
-      <ul className="list">
-        {activeChats?.map((chat) => (
-          <ChatItem
-            key={chat.id}
-            name={chat.supporteeName || 'מחפשים...'}
-            isEnded={false}
-            chatId={chat.id}
-          />
-        ))}
-        {endedChats && endedChats.length > 0 ? (
-          <>
-            <p className="chats-ended">שיחות שהסתיימו</p>
-            {endedChats?.map((chat) => (
+  useEffect(() => {
+    if (!user) return;
+
+    // todo: make more generic
```

Comment:
In line 28:
`    let role: Role = "supporter";
`
For future perposes, if we want the "supportee" to have a chatsList page as well, the `role` variable should be generic instead of hard-coded.